datacompy/core.py

```diff
@@ -792,6 +797,15 @@ def columns_equal(
         values don't match.
     """
     compare: pd.Series[bool]
+
+    # short circuit if comparing mixed type columns. We don't want to support this moving forward.
+    if pd.api.types.infer_dtype(col_1).startswith("mixed") or pd.api.types.infer_dtype(
+        col_1
+    ).startswith("mixed"):
```

Comment:
is this supposed to be `pd.api.types.infer_dtype(col_2).startswith("mixed")` instead for the second clause?